src/net/messages.rs

```diff
@@ -0,0 +1,124 @@
+use heapless::{consts, String, Vec};
+use serde::Serialize;
+
+use core::fmt::Write;
+
+#[derive(Debug, Copy, Clone)]
+pub enum SettingsResponseCode {
+    NoError = 0,
+    NoTopic = 1,
+    InvalidPrefix = 2,
+    UnknownTopic = 3,
+    UpdateFailure = 4,
+}
+
+/// Represents a generic MQTT message.
+pub struct MqttMessage<'a> {
```

Comment:
As per discussion, for now, this will live in stabilizer, but we will likely copy this all over to `miniconf` at a later date behind an MQTT-feature flag.